clang/lib/Headers/amdgpuintrin.h

```diff
@@ -0,0 +1,155 @@
+//===-- amdgpuintrin.h - AMDPGU intrinsic functions -----------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __AMDGPUINTRIN_H
+#define __AMDGPUINTRIN_H
+
+#ifndef __AMDGPU__
+#error "This file is intended for AMDGPU targets or offloading to AMDGPU"
+#endif
+
+#include <stdbool.h>
+#include <stdint.h>
```

Comment:
I think it's a bit surprising that including this header will transitively define `bool` in C modes; I would expect we'd use `_Bool` in this header file.

As for C89 modes, I think this will still work okay, but you should have explicit test coverage for C89.

Reply:
Hrmmm. My gut feeling is: this should be clean for C89 and just work because there's no reason for it not to. Alternatively, if it shouldn't work in C89, perhaps we should explicitly disallow that mode?

Reply:
Alright, guess I could make it work.

Reply:
Done, and a test.